Upgrade xcode to 9.4 on v0.14.x branch

hmaarrfk opened the pull request:
## Description
Should be merged in conjunction with https://github.com/scikit-image/scikit-image/pull/3723

## Checklist

<!-- It's fine to submit PRs which are a work in progress! -->
<!-- But before they are merged, all PRs should provide: -->
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

<!-- For detailed information on these and other aspects see -->
<!-- the scikit-image contribution guidelines. -->
<!-- https://scikit-image.org/docs/dev/contribute.html -->

## For reviewers

<!-- Don't remove the checklist below. -->
- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

hmaarrfk commented:
Why is there no module named dask again???? https://travis-ci.org/scikit-image/scikit-image/jobs/488780690#L6560

stefanv commented:
I guess because `dask` is now an optional requirement and has to be installed explicitly.

hmaarrfk commented:
Right, for a while sphinx related dependencies on OSX were broken so @jni and I fixed it by having a special OSX Switch.

this commit I made a while back
https://github.com/scikit-image/scikit-image/commit/3074a005b1164682216e297622c3d3ec60d3971a

Reverted that change to make OSX the same as linux. I just backported it. I think it should pass tests now.

I almost feel like blanket replacing the build system in v0.14.x with the one in master.
